app/frontend/src/features/connections/friends/useCancelFriendRequest.ts

```diff
@@ -21,10 +22,10 @@ export default function useCancelFriendRequest() {
     {
       onMutate: async ({ setMutationError }) => {
         setMutationError("");
-        await queryClient.cancelQueries("friendRequestsSent");
       },
-      onSuccess: () => {
+      onSuccess: (_, { userId }) => {
         queryClient.invalidateQueries("friendRequestsSent");
+        queryClient.invalidateQueries(["user", userId]);
```

Comment:
Needed this otherwise when you cancel a friend request then go back to the user's page, the "add friend" button won't reappear as the friendship status on the user query would be stale.